common/libzkp/impl/src/utils.rs

```diff
@@ -11,6 +11,19 @@ use std::{
 pub(crate) static OUTPUT_DIR: Lazy<Option<String>> =
     Lazy::new(|| env::var("PROVER_OUTPUT_DIR").ok());
 
+/// # Safety
+#[no_mangle]
+pub extern "C" fn free_c_chars(ptr: *mut c_char) {
+    if ptr.is_null() {
+        log::warn!("Try to free an empty pointer!");
+        return;
+    }
+
+    unsafe {
+        let _ = CString::from_raw(ptr);
```

Comment:
`This should only ever be called with a pointer that was earlier obtained by calling [CString::into_raw](https://doc.rust-lang.org/stable/std/ffi/struct.CString.html#method.into_raw).`

does the ptr is obtained by `into_raw`

Reply:
Yes, the proof results and VKs are all returned by `CString::into_raw` through [vec_to_c_char](https://github.com/scroll-tech/scroll/blob/develop/common/libzkp/impl/src/utils.rs#L29C34-L29C42) and [string_to_c_char](https://github.com/scroll-tech/scroll/blob/develop/common/libzkp/impl/src/utils.rs#L25C35-L25C43).

e.g. [the result of chunk-proof](https://github.com/scroll-tech/scroll/blob/develop/common/libzkp/impl/src/chunk.rs#L96)